diff -urN vbam-1.8.0.1054/fex/fex/Data_Reader.cpp vbam-1.8.0.1054-patched/fex/fex/Data_Reader.cpp
--- vbam-1.8.0.1054/fex/fex/Data_Reader.cpp	2011-12-05 02:14:43.000000000 +1100
+++ vbam-1.8.0.1054-patched/fex/fex/Data_Reader.cpp	2012-09-23 10:09:15.650572579 +1100
@@ -732,11 +732,11 @@
 
 blargg_err_t Gzip_File_Reader::read_v( void* p, int s )
 {
-	int result = gzread( file_, p, s );
+	int result = gzread( (gzFile)file_, p, s );
 	if ( result != s )
 	{
 		if ( result < 0 )
-			return convert_gz_error( file_ );
+			return convert_gz_error( (gzFile)file_ );
 		
 		return blargg_err_file_corrupt;
 	}
@@ -746,8 +746,8 @@
 
 blargg_err_t Gzip_File_Reader::seek_v( int n )
 {
-	if ( gzseek( file_, n, SEEK_SET ) < 0 )
-		return convert_gz_error( file_ );
+	if ( gzseek( (gzFile)file_, n, SEEK_SET ) < 0 )
+		return convert_gz_error( (gzFile)file_ );
 
 	return blargg_ok;
 }
@@ -756,7 +756,7 @@
 {
 	if ( file_ )
 	{
-		if ( gzclose( file_ ) )
+		if ( gzclose( (gzFile)file_ ) )
 			check( false );
 		file_ = NULL;
 	}
diff -urN vbam-1.8.0.1054/src/Util.cpp vbam-1.8.0.1054-patched/src/Util.cpp
--- vbam-1.8.0.1054/src/Util.cpp	2011-05-25 22:55:46.000000000 +1100
+++ vbam-1.8.0.1054-patched/src/Util.cpp	2012-09-23 10:09:21.210572537 +1100
@@ -590,7 +590,7 @@
 
 gzFile utilGzOpen(const char *file, const char *mode)
 {
-  utilGzWriteFunc = (int (ZEXPORT *)(void *,void * const, unsigned int))gzwrite;
+  utilGzWriteFunc = (int (ZEXPORT *)(gzFile, void * const, unsigned int))gzwrite;
   utilGzReadFunc = gzread;
   utilGzCloseFunc = gzclose;
   utilGzSeekFunc = gzseek;
